lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetSchemaConverter.java

```diff
@@ -50,18 +50,32 @@
 import static io.trino.spi.type.TinyintType.TINYINT;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
+import static org.apache.parquet.schema.LogicalTypeAnnotation.decimalType;
 import static org.apache.parquet.schema.Type.Repetition.OPTIONAL;
 
 public class ParquetSchemaConverter
 {
+    // Map precision to the number bytes needed for binary conversion.
+    // Based on org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe
+    private static final int[] PRECISION_TO_BYTE_COUNT = new int[38];
+
+    static {
+        for (int precision = 1; precision <= 38; precision++) {
+            // Estimated number of bytes needed.
+            PRECISION_TO_BYTE_COUNT[precision - 1] = (int) Math.ceil((Math.log(Math.pow(10, precision) - 1) / Math.log(2) + 1) / 8);
```

Comment:
wouldn't it be clearer to use `PRECISION_TO_BYTE_COUNT[precision]` (`precision` as the index, without decrementing)?
(this would require adding one more entry to the array)